### PR TITLE
[SPARK-39050][SQL] Error class: UNSUPPORTED_OPERATION to UNSUPPORTED_FEATURE

### PR DESCRIPTION
**Pull request by srielau**
### What changes were proposed in this pull request?

UNSUPPORTED_OPERATION will be removed and replaced with the existing UNSUPPORTED_FEATURE.
This effects three errors:
ARROW TIMESTAMP, ORC TMESTAMP TO TIMESTAMP_NTZ and  ORC TMESTAMP_NTZ TO TIMESTAMP

### Why are the changes needed?

Clean up ERROR CLASSES before publishing them.

### Does this PR introduce _any_ user-facing change?

No, this is still internal, unreleased  code

### How was this patch tested?

Run existing QueryExecutionErrorsSuite

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 1]
```diff
@@ -261,41 +261,45 @@ class QueryExecutionErrorsSuite
     }
   }
 
-  test("UNSUPPORTED_OPERATION: timeZoneId not specified while converting TimestampType to Arrow") {
+  test("UNSUPPORTED_FEATURE: timeZoneId not specified while converting TimestampType to Arrow") {
     checkErrorClass(
       exception = intercept[SparkUnsupportedOperationException] {
         ArrowUtils.toArrowSchema(new StructType().add("value", TimestampType), null)
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:267]: I wonder is it possible to trigger the error from user space? If not, let use `INTERNAL_ERROR` or `IllegalStateException`.

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:267]: Let's replace the check https://github.com/apache/spark/blob/88e53e59e746a6d3aadac3cdc5fcbd74d90ee0e9/sql/catalyst/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala#L51 by an assert, and remove the test. `ArrowUtils.toArrowType()` is internal API, BTW.

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -192,6 +192,9 @@
       "AES_MODE" : {
         "message" : [ "AES-<mode> with the padding <padding> by the <functionName> function." ]
       },
+      "ARROW_TIMESTAMP_WITHOUT_TIMEZONE" : {
+        "message" : [ "TIMESTAMP must supply timeZoneId parameter while converting to the arrow timestamp type." ]
+      },
```
[the remark this answers is not in the document]
Comment (srielau) [reply core/src/main/resources/error/error-classes.json]: Oops... also need to fix the arrow test suite...

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 3]
```diff
@@ -1616,10 +1621,8 @@ object QueryExecutionErrors extends QueryErrorsBase {
 
   def timeZoneIdNotSpecifiedForTimestampTypeError(): Throwable = {
```
[the remark this answers is not in the document]
Comment (srielau) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala]: Removed

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @srielau and @cloud-fan for review.

**Pull request closed without merging**